src/storage/tuple_accessor.cpp

```diff
@@ -0,0 +1,222 @@
+#include "storage/tuple_accessor.h"
+
+#include <llvm/ExecutionEngine/ExecutionEngine.h>
+#include <llvm/ExecutionEngine/MCJIT.h>
+#include <llvm/IR/DerivedTypes.h>
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/IR/Module.h>
+#include <llvm/IR/PassManager.h>
+#include <llvm/IR/Verifier.h>
+
+#include "common/constants.h"
+#include "common/llvm.h"
+#include "storage/data_table.h"
+#include "storage/storage_util.h"
+#include "transaction/transaction_context.h"
+
+namespace terrier::storage {
+
+static const char INSERT_INTO_NAME[] = "insert_into";
+
+TupleAccessor::TupleAccessor(DataTable *datatable) {
+  auto module = std::make_unique<llvm::Module>("tuple_accessor", context_);
+
+  BuildInsertInto(module.get(), datatable);
+
+  llvm::ModuleAnalysisManager mam;
+  llvm::VerifierAnalysis verifier;
+  auto UNUSED_ATTRIBUTE result = verifier.run(*module, mam);
+  TERRIER_ASSERT(!result.IRBroken, "Compilation must not fail");
+
+  std::string err;
+  engine_ = std::unique_ptr<llvm::ExecutionEngine>(llvm::EngineBuilder(std::move(module))
+                                                       .setErrorStr(&err)
+                                                       .setOptLevel(llvm::CodeGenOpt::Aggressive)
+                                                       .setEngineKind(llvm::EngineKind::JIT)
+                                                       .create());
+  if (engine_ == nullptr) {
+    throw std::runtime_error(err);
+  }
+  TERRIER_ASSERT(engine_ != nullptr, "Engine initialization should not fail");
+
+  datatable->InsertInto = (void (*)(terrier::common::ManagedPointer<terrier::transaction::TransactionContext>,
+                                    const terrier::storage::ProjectedRow &,
+                                    terrier::storage::TupleSlot))engine_->getFunctionAddress(INSERT_INTO_NAME);
+}
+
+void TupleAccessor::BuildInsertInto(llvm::Module *module, DataTable *datatable) {
```

Comment:
This is what I think needs the most attention.  If this were ASM, I'd have a comment on every line explaining what that line is doing in addition to guideposts.  I am thinking that is equally appropriate here, but I wanted to get input before investing that level of energy.